docs/source/deployguide/ca-config.md

```diff
@@ -0,0 +1,417 @@
+- **csr.pathlength**: If you will have an intermediate CA, set this value to `1` in the root CA. If this is an intermediate CA the value would be `0`.
+
+## signing
+
+```
+signing:
+    default:
+      usage:
+        - digital signature
+      expiry: 8760h
+    profiles:
+      ca:
+         usage:
+           - cert sign
+           - crl sign
+         expiry: 43800h
+         caconstraint:
+           isca: true
+           maxpathlen: 0
+      tls:
+         usage:
+            - signing
+            - key encipherment
+            - server auth
+            - client auth
+            - key agreement
+         expiry: 8760h
+```
+
+The defaults in this section are normally sufficient for a production server. However, you might want to modify the default expiration of the generated enrollment CA and TLS certificates. Note that this is different than the `expiry` specified in the `csr` section for the CA root certificate.  
+
+If this is a TLS CA, it is recommended that you remove the `ca` section from `profiles:` since a TLS CA should only be issuing TLS certificates.
+
+If you plan to have an intermediate CA in addition to the root CA, then you must set  `maxpathlength` to `1` in the configuration .yaml file for the root CA. This field represents the maximum number of non-self-issued intermediate certificates that can follow this certificate in a certificate chain.
+
+## bccsp
+
+```
+bccsp:
+    default: SW
+    sw:
+        hash: SHA2
+        security: 256
+        filekeystore:
+            # The directory used for the software file-based keystore
+            keystore: msp/keystore
+```
+
+If you plan to use a [Hardware Security Module](https://hyperledger-fabric-ca.readthedocs.io/en/release-1.4/users-guide.html#example) (HSM) you need to configure the `hash` and `security` of the HSM crypto library. When certificates are generated, the private key is generated by and stored in the HSM instead of the `msp/keystore` folder.
+
+
+## cors
+```
+cors:
+    enabled: false
+    origins:
+      - "*"
+```
+
+Cross-Origin Resource Sharing (CORS) can be configured to to use additional HTTP headers to tell browsers to give a web application running at one origin access to selected resources from a different origin.
+
+## cfg
+```
+cfg:
+  affiliations:
+    allowremove: false
+  identities:
+    allowremove: false  
+```
+
+These two parameters are not listed in the sample configuration file, but are important to understand. With the default configuration set to false, you will be unable to remove affiliations or identities without a server restart. If you anticipate the need to remove affiliations or identities from your production environment without a server restart, both of these fields should be set to `true` before starting your server. Note that after the server is started, affiliations and identities can only be modified by using the Fabric CA client CLI commands.  
```

Comment:
Should this section be linked to from the affiliations section given the dependencies between them?

Reply:
Excellent suggestion. Thank you.